docs/proposals/external_providers_grpc.md

```diff
@@ -54,3 +54,22 @@ https://github.com/gardener/machine-controller-manager/tree/grpc-driver
 
 Refer GRPC based implementation for AWS client:
 https://github.com/ggaurav10/aws-driver-grpc
+
+## Alternatives:
+### Cloud Provider as GRPC server:
+One approach would be to make MCM as a GRPC client and Cloud Provider as a GRPC server.
+Here the Cloud Provider will provide the following GRPC services:
+1. `Create()` creates a VM
+1. `Delete()` deletes a VM
+1. `List()` lists VMs
+
+In this approach also, a GRPC stream would be required so that the CP can request for machine class and secret from MCM using the functions `GetMachineClass()` and `GetSecret()`.
```

Comment:
Does not have to be a stream (it can be, though). It can be individual calls as well.

It is not clear from this description that the MCM will need to be the server for the `GetMachineClass` and `GetSecret` operations.